File: CS/MasterDetailProject/CustomGridControl/CustomGridView.cs
using CustomGridControl;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Tab;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraTab.ViewInfo;
using MasterDetailProject.CustomGridControl;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;

namespace MasterDetailProject {
    [System.ComponentModel.DesignerCategory("")]
    class CustomGridView : GridView {
        public CustomGridView() {
        }

        public CustomGridView(GridControl grid)
            : base(grid) {
            FieldInfo[] fields = (base.TabControl).GetType().GetFields(
                         BindingFlags.NonPublic |
                         BindingFlags.Instance);

            FieldInfo fInfo = fields.First(item => item.Name == "viewInfo");

            fInfo.SetValue(this.TabControl, new CustomBaseTabControlViewInfo(this.TabControl));
        }

        protected override string ViewName {
            get {
                return "CustomGridView";
            }
        }

    }

}

File: VB/MasterDetailProject/CustomGridControl/CustomGridView.vb
Imports Microsoft.VisualBasic
Imports CustomGridControl
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Tab
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraTab.ViewInfo
Imports MasterDetailProject.CustomGridControl
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Linq
Imports System.Reflection
Imports System.Text
Imports System.Threading.Tasks

Namespace MasterDetailProject
	<System.ComponentModel.DesignerCategory("")> _
	Friend Class CustomGridView
		Inherits GridView
		Public Sub New()
		End Sub

		Public Sub New(ByVal grid As GridControl)
			MyBase.New(grid)
			Dim fields() As FieldInfo = (MyBase.TabControl).GetType().GetFields(BindingFlags.NonPublic Or BindingFlags.Instance)

			Dim fInfo As FieldInfo = fields.First(Function(item) item.Name = "viewInfo")

			fInfo.SetValue(Me.TabControl, New CustomBaseTabControlViewInfo(Me.TabControl))
		End Sub

		Protected Overrides ReadOnly Property ViewName() As String
			Get
				Return "CustomGridView"
			End Get
		End Property

	End Class

End Namespace
